### PR TITLE
[APMSP-1458] Fix single-span sampling

### PR DESCRIPTION
**Pull request by VianneyRuhlmann**
# What does this PR do?

Several fixes to sampling:
- Single-span sampling should not be made based on priority but based on sampling mechanism
- Analyzed tag is `_dd1.sr.eausr`
- Use const for keys

# Motivation

What inspired you to submit this pull request?

# Additional Notes

Anything else we should know when reviewing?

# How to test the change?

Describe here in detail how the change can be validated.

### CODE REVIEW

File: data-pipeline/src/trace_exporter.rs [hunk 1]
```diff
@@ -98,25 +103,28 @@ fn add_path(url: &Uri, path: &str) -> Uri {
...
-                    // We send chunks with positive priority
-                    return true;
-                }
-                // We send single spans with positive priority
-                sampled_indexes.push(index);
-            } else if priority.is_none() && has_top_level(span) {
-                // We send chunks with no priority
+            // PrioritySampler and NoPrioritySampler
+            let priority = span.metrics.get(SAMPLING_PRIORITY_KEY);
+            if has_top_level(span) && (priority.is_none() || priority.is_some_and(|p| *p > 0.0)) {
```
Comment (ekump) [inline data-pipeline/src/trace_exporter.rs:116]: minor: Do you think `priority.map_or(true, |p| *p > 0.0)` is clearer than `(priority.is_none() || priority.is_some_and(|p| *p > 0.0))`?

Comment (ekump) [reply data-pipeline/src/trace_exporter.rs:116]: Also, do we have sufficient unit tests covering this logic here? I didn't see any changes to tests.

**Review by ekump — APPROVED** [review]
a couple of minor comments, but LGTM

File: data-pipeline/src/trace_exporter.rs (hunk 1 above)
Comment (VianneyRuhlmann) [reply data-pipeline/src/trace_exporter.rs:116]: There are tests [here](https://github.com/DataDog/libdatadog/blob/321e98ecd1be000873ec1ba8831977cc6787683f/data-pipeline/src/trace_exporter.rs#L989). The sampling logic has not changed so no test update required.

The `(priority.is_none() || priority.is_some_and(|p| *p > 0.0))` syntax seems clearer to me

**Pull request merged**